pkg/kubelet/kubelet.go

```diff
@@ -3282,7 +3311,21 @@ func GetPhase(spec *api.PodSpec, info []api.ContainerStatus) api.PodPhase {
 
 func (kl *Kubelet) generateAPIPodStatus(pod *api.Pod, podStatus *kubecontainer.PodStatus) api.PodStatus {
 	glog.V(3).Infof("Generating status for %q", format.Pod(pod))
+
+	// any pod that requires eviction must be added to the list to sync.
+	attrs := &lifecycle.PodEvictAttributes{Pod: pod}
+	for _, podEvictor := range kl.podEvictors {
+		if result := podEvictor.Evict(attrs); result.Evict {
```

Comment:
edited my above comment, `Evict` will be called on each call to generateAPIPodStatus, but its not possible for us to return true one time and false another if we add the `SyncPod` method.  In general, I am not sure its worth the split right now.

I think I want to just update this to have `syncPod` check for the failed status and immediately kill the pod, and then call it a day and we iterate as we integrate the other evictors.